packages/flame/lib/src/sprite_batch.dart

```diff
@@ -162,7 +187,11 @@ class SpriteBatch {
   }
 
   /// The atlas used by the [SpriteBatch].
-  Image atlas;
+  final Image atlas;
+
+  /// The horizontally flipped atlas generated by the [atlas].
+  /// This is available when there is flipped transform added by [addTransform].
+  Image? flippedAtlas;
```

Comment:
Can't there be two sprite batch objects instead of duplicating all the fields in here?

Reply:
That's a good idea but it narrows the use cases to tiled only and needs to remove `flip` from `addTransform` because the `SpriteBatch` does not have ability to render flipped `BatchItem`. 
https://github.com/flame-engine/flame/blob/e5345e62d0f1cdf96669141f5e22fe7bf7224a39/packages/flame/lib/src/sprite_batch.dart#L253-L258

Reply:
@spydon 
I am going to switch implementation based on your opinion.

It would have a break changes to users who are using `SpriteBatch` directly with flipped transform.
1. No more `flip` property on `SpriteBatch.addTransform()`
2. No more rendering via drawImageRect before flipped atlas is ready because the atlas should be made before construction.

I agree with creating another SpriteBatch instead of internally supporting it by duplicating lots of properties. And IMHO, there are few users who use `SpriteBatch` directly with a flip.

Does breaking those changes are acceptable?

Reply:
Sounds good to me!